extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -129,54 +145,92 @@ static io.deephaven.qst.type.Type<?> convertPrimitiveType(@NotNull final Type ic
+    /**
+     * List all {@link Namespace namespaces} in a given namespace. This method is only supported if the catalog
+     * implements {@link SupportsNamespaces} for namespace discovery. See
+     * {@link SupportsNamespaces#listNamespaces(Namespace)}.
+     *
+     * @param namespace The namespace to list namespaces in.
+     * @return A list of all namespaces in the given namespace.
+     */
+    public List<Namespace> listNamespaces(@NotNull final Namespace namespace) {
+        if (catalog instanceof SupportsNamespaces) {
             final SupportsNamespaces nsCatalog = (SupportsNamespaces) catalog;
             return nsCatalog.listNamespaces(namespace);
         }
         throw new UnsupportedOperationException(String.format(
                 "%s does not implement org.apache.iceberg.catalog.SupportsNamespaces", catalog.getClass().getName()));
     }
 
+    /**
+     * List all {@link Namespace namespaces} in the catalog as a Deephaven {@link Table table}. The resulting table will
+     * be static and contain the same information as {@link #listNamespaces()}.
+     *
+     * @return A {@link Table table} of all namespaces.
+     */
     public Table listNamespacesAsTable() {
         return listNamespacesAsTable(Namespace.empty());
     }
 
-    public Table listNamespacesAsTable(@NotNull Namespace namespace) {
-        final List<Namespace> namespaces = listNamespaces();
+    /**
+     * List all {@link Namespace namespaces} in a given namespace as a Deephaven {@link Table table}. The resulting
+     * table will be static and contain the same information as {@link #listNamespaces(Namespace)}.
+     *
+     * @return A {@link Table table} of all namespaces.
+     */
+    public Table listNamespacesAsTable(@NotNull final Namespace namespace) {
+        final List<Namespace> namespaces = listNamespaces(namespace);
         final long size = namespaces.size();
 
         // Create and return a table containing the namespaces as strings
         final Map<String, ColumnSource<?>> columnSourceMap = new LinkedHashMap<>();
 
         // Create the column source(s)
-        final WritableColumnSource<String> namespaceColumn =
-                ArrayBackedColumnSource.getMemoryColumnSource(size, String.class, null);
-        columnSourceMap.put("namespace", namespaceColumn);
+        final String[] namespaceArr = new String[(int) size];
+        columnSourceMap.put("namespace", InMemoryColumnSource.getImmutableMemoryColumnSource(namespaceArr));
 
-        final WritableColumnSource<Object> objectColumn =
-                ArrayBackedColumnSource.getMemoryColumnSource(size, Object.class, Namespace.class);
-        columnSourceMap.put("namespace_object", objectColumn);
+        final Namespace[] namespaceObjectArr = new Namespace[(int) size];
+        columnSourceMap.put("namespace_object",
+                InMemoryColumnSource.getImmutableMemoryColumnSource(namespaceObjectArr, Namespace.class, null));
```

Comment:
I'd prefer you either always be explicit about data type and component type, or never.
```suggestion
                InMemoryColumnSource.getImmutableMemoryColumnSource(namespaceObjectArr);
```